deep_qa/layers/backend/squeeze.py

```diff
@@ -26,6 +26,11 @@ def compute_mask(self, inputs, mask=None):
         # pylint: disable=unused-argument
         if mask is None:
             return None
+        # If the mask doesn't need squeezing, just return it.
+        # TODO(becky): better to check if the last mask dim == 1?
+        if K.ndim(inputs) == K.ndim(mask) + 1:
```

Comment:
i need it for the tuple_inference: https://github.com/allenai/deep_qa/blob/master/deep_qa/models/multiple_choice_qa/tuple_inference.py#L163-L164
At least when I was using the EncoderWrappers (not TimeDistributedWithMask). the resulting mask at the end of the matching was one dimension shorter and so didn't need squeezing (and using squeeze made it crash bc of Squeeze.compute_mask).  That said, if your change to TimeDistributedWithMask makes the dimensions of the tensor and its mask identical (either way, both with the extra dim of size 1 or both without it) then it wouldn't be necessary.